Source/Scene/GroundPolygon.js

```diff
@@ -0,0 +1,424 @@
+
+            EncodedCartesian3.writeElements(position, vbPositions, index);
+            EncodedCartesian3.writeElements(position, vbPositions, index + 6);
+            index += 12;
+
+            Cartesian3.pack(Cartesian3.ZERO, vbNormals, normalIndex);
+            Cartesian3.pack(normal, vbNormals, normalIndex + 3);
+            normalIndex += 6;
+        }
+
+        var numWalls = walls.length;
+        var wallPositions;
+        var wallLength;
+
+        for (i = 0; i < numWalls; ++i) {
+            wall = walls[i];
+            wallPositions = wall.attributes.position.values;
+            wallLength = wallPositions.length / 2;
+
+            for (j = 0; j < wallLength; j += 3) {
+                position = Cartesian3.unpack(wallPositions, j, scratchPosition);
+                ellipsoid.scaleToGeodeticSurface(position, position);
+
+                normal = ellipsoid.geodeticSurfaceNormal(position, scratchNormal);
+                Cartesian3.multiplyByScalar(normal, upDelta, normal);
+
+                EncodedCartesian3.writeElements(position, vbPositions, index);
+                EncodedCartesian3.writeElements(position, vbPositions, index + wallLength * 2);
+                index += 6;
+
+                Cartesian3.pack(normal, vbNormals, normalIndex);
+                Cartesian3.pack(Cartesian3.ZERO, vbNormals, normalIndex + wallLength);
+                normalIndex += 3;
+            }
+
+            index += wallLength * 2;
+            normalIndex += wallLength;
+        }
+
+        var numIndices = numCapIndices + numWallIndices;
+        var ibIndices = IndexDatatype.createTypedArray(numVertices, numIndices);
+
+        var i0;
+        var i1;
+        var i2;
+
+        index = 0;
+        for (i = 0; i < numBottomIndices; i += 3) {
+            i0 = bottomIndices[i] * 2;
+            i1 = bottomIndices[i + 1] * 2;
+            i2 = bottomIndices[i + 2] * 2;
+
+            ibIndices[index++] = i2;
+            ibIndices[index++] = i1;
+            ibIndices[index++] = i0;
+        }
+
+        for (i = 0; i < numBottomIndices; i += 3) {
+            i0 = bottomIndices[i] * 2;
+            i1 = bottomIndices[i + 1] * 2;
+            i2 = bottomIndices[i + 2] * 2;
+
+            ibIndices[index++] = i0 + 1;
+            ibIndices[index++] = i1 + 1;
+            ibIndices[index++] = i2 + 1;
+        }
+
+        var offset = numCapVertices;
+        for (i = 0; i < numWalls; ++i) {
+            wall = walls[i];
+            var wallIndices = wall.indices;
+            wallLength = wallIndices.length;
+
+            for (j = 0; j < wallLength; ++j) {
+                ibIndices[index++] = wallIndices[j] + offset;
+            }
+            offset += wall.attributes.position.values.length / 3;
+        }
+
+        var positionBuffer = context.createVertexBuffer(vbPositions, BufferUsage.STATIC_DRAW);
+        var normalBuffer = context.createVertexBuffer(vbNormals, BufferUsage.STATIC_DRAW);
```

Comment:
What are the normals used for?  Just for extruding the bottom in the vertex shader?

We shouldn't need them for lighting since we are just going to modulate the terrain color - and they are wrong given the terrain.